### DISCUSSION TITLE
Can I limit by special ip?

### DISCUSSION THREAD

**Discussion by ormon-saadanbek** [jsocol/django-ratelimit#238] [Q&A]
Hi there!
Can I limit user by special ip, cause I have a form where only 3-5 endless using it, I want to limit them.
How I can do it?

**Comment by jsocol** [comment]
Yes! You can do this with [callable rates](https://django-ratelimit.readthedocs.io/en/latest/rates.html#callables). The exact implementation depends on the details of what you’re looking for but you could do

```py
LIMIT_IPS = set([ip1, ip2, …])

def special_rate(group, request):
    if request.META['REMOTE_ADDR'] in LIMIT_IPS:
        return '5/h'  # limit for these IPs
    # for everyone else
    return None  # None for no limit, or 0 for no requests allowed, or another rate

@ratelimit(key='ip', rate=special_rate)
def my_view(request):
    # ….
```
